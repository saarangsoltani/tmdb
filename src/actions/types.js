export const API = "api";
export const API_START = "api_start";
export const API_END = "api_end";
export const SET_DISCOVERED_SERIES = "set_series";
export const SORT_SERIES = "sort_series";
export const FILTER_BY_GENRES = "filter_by_genres";
