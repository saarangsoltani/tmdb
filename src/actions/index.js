import {
  API,
  SET_DISCOVERED_SERIES,
} from "./types";

export const discoverSeries = () => ({
  type: API,
  payload: {
    url: "discover/tv",
    onSuccess: setDiscoveredSeriesData
  }
});

export const setDiscoveredSeriesData = data => ({
  type: SET_DISCOVERED_SERIES,
  payload: data
});
