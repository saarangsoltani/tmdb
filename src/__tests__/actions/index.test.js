import * as actions from "../../actions";
import { API, SET_DISCOVERED_SERIES } from "../../actions/types";

describe("actions/discover", () => {
  it("discoverSeries", () => {
    expect(actions.discoverSeries()).toMatchSnapshot();
  });

  it("setDiscoveredSeriesData", () => {
    expect(actions.setDiscoveredSeriesData("data")).toMatchSnapshot();
  });
});
