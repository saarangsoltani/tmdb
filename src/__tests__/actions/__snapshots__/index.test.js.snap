// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actions/discover discoverSeries 1`] = `
Object {
  "payload": Object {
    "onSuccess": [Function],
    "url": "discover/tv",
  },
  "type": "api",
}
`;

exports[`actions/discover setDiscoveredSeriesData 1`] = `
Object {
  "payload": "data",
  "type": "set_series",
}
`;
